src/network/networktool.cpp

```diff
@@ -0,0 +1,181 @@
+#include "logger.h"
+#include "maddy/parser.h"
+#include "networkmanager.h"
+#include "networktool.h"
+#include "notificationmanager.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QRegularExpression>
+#include <QTimer>
+#include <QUrl>
+
+PING_LOGGING_CATEGORY(NETWORKTOOL, "ping.networktool")
+
+void NetworkTool::checkInterfaceUpdate()
+{
+    // It's only necessary to get the gitUserRepo variable only once
+    static bool firstTime = true;
+    static QString gitUserRepo;
+    if(firstTime) {
+        //https://github.com/user/repo.git
+        const static QRegularExpression regex("github.com/(.*).git");
+        QRegularExpressionMatch regexMatch = regex.match(QStringLiteral(GIT_URL));
+        // TODO: Check regex output
+        if (regexMatch.hasMatch()) {
+            gitUserRepo = regexMatch.capturedTexts()[1];
+        } else {
+            qCWarning(NETWORKTOOL) <<
+                                   "Fail to get github user and repository! It'll not be possible to check for updates";
+            return;
+        }
+        firstTime = false;
+    }
+
+    const QUrl url{QStringLiteral("https://api.github.com/repos/%1/releases").arg(gitUserRepo)};
```

Comment:
what happens if the first part failed and it gets here? fails silently?

Reply:
What part ? Are you talking about `QStringLiteral` ?

Reply:
it still makes the request even with gitUserRepo empty?
nevermind, didn't notice the return there.

Reply:
I meant it could try to make requests without gitUserRepo set. looks fine now.